pkg/controllers/mcs/service_export_controller.go

```diff
@@ -408,6 +419,10 @@ func (c *ServiceExportController) reportEndpointSliceWithServiceExportCreate(ctx
 		return nil
 	}
 
+	err = c.waitTargetResourceSyncedOnCluster(ctx, serviceExportKey.Cluster, time.Second, endpointSliceGVR)
+	if err != nil {
+		return err
+	}
```

Comment:
```suggestion
	// Before retrieving EndpointSlice objects from the informer, ensure the informer cache is synced.
	// This is necessary because the informer for EndpointSlice is created dynamically in the Reconcile() routine
	// when a Work resource containing an EndpointSlice is detected for the cluster. If the informer is not yet synced,
	// we wait for the cache to be ready to avoid reading incomplete or missing data.
	if !singleClusterManager.IsInformerSynced(endpointSliceGVR) {
		err = c.waitForResourceSyncOnCluster(ctx, endpointSliceGVR, serviceExportKey.Cluster)
		if err != nil {
			return err
		}
	}
```

Reply:
I prefer the 1st approach I mentioned above. Please consider how much time it would hanging here waiting for cache sync.